rust/Cargo.toml

```diff
@@ -42,23 +42,24 @@ lance-linalg = { version = "=0.8.3", path = "./lance-linalg" }
 lance-testing = { version = "=0.8.3", path = "./lance-testing" }
 lance-test-macros = { version = "=0.8.3", path = "./lance-test-macros" }
 # Note that this one does not include pyarrow
-arrow = { version = "43.0.0", optional = false }
-arrow-arith = "43.0"
-arrow-array = "43.0"
-arrow-buffer = "43.0"
-arrow-cast = "43.0"
-arrow-data = "43.0"
-arrow-ord = "43.0"
-arrow-row = "43.0"
-arrow-schema = "43.0"
-arrow-select = "43.0"
+arrow = { version = "46.0.0", optional = false }
+arrow-arith = "46.0"
```

Comment:
IIRC, we are wait for arrow 47 for a few fixes? And datafusion 32 is being voted at the moment, right?

Reply:
Why wait?

Reply:
Do we need to bump `datafusion` again in 3-5 days to include fixes?

Reply:
🤷 yes, we need to upgrade for the next version. IMO we should upgrade on every datafusion release.